ethereum/test/shutdownswitch.js

```diff
@@ -0,0 +1,528 @@
+        assert.equal((await initialized.methods.requiredVotesToShutdown().call()), 3)
+    })
+    
+    it("should reject a vote from a guardian that is not active", async function () {
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+
+        // Cast a vote using a valid auth proof but for a non-active guardian, and it should fail.
+        let voteFailed = false;
+        try {
+            await initialized.methods.castShutdownVote(GuardianAuthProof9).send({ from: EthPublicKey });
+        } catch (error) {
+            assert.equal(error.message, "Returned error: VM Exception while processing transaction: revert you are not a registered voter")
+            voteFailed = true
+        }
+
+        assert.ok(voteFailed)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+        assert.equal((await initialized.methods.numVotesToShutdown().call()), 0)
+        assert.equal((await initialized.methods.requiredVotesToShutdown().call()), 3)
+    })
+        
+    it("should reject a vote cast using the wrong wallet", async function () {
+        const accounts = await web3.eth.getAccounts();
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+
+        // Cast a vote using a valid auth proof but for a non-active guardian, and it should fail.
+        let voteFailed = false;
+        try {
+            await initialized.methods.castShutdownVote(GuardianAuthProof0).send({ from: accounts[0] });
+        } catch (error) {
+            assert.equal(error.message, "Returned error: VM Exception while processing transaction: revert you are not a registered voter")
+            voteFailed = true
+        }
+
+        assert.ok(voteFailed)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+        assert.equal((await initialized.methods.numVotesToShutdown().call()), 0)
+        assert.equal((await initialized.methods.requiredVotesToShutdown().call()), 3)
+    })
+
+    it("should take three guardian votes to disable transfers", async function () {
+        const accounts = await web3.eth.getAccounts();
+        const initialized = new web3.eth.Contract(BridgeImplementationFullABI, TokenBridge.address);
+        await clearAllVotes(initialized);
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+
+        // This first vote should succeed, but we should still be enabled.
+        await initialized.methods.castShutdownVote(GuardianAuthProof0).send({ from: EthPublicKey });
+
+        assert.equal((await initialized.methods.numVotesToShutdown().call()), 1)
+        assert.equal((await initialized.methods.requiredVotesToShutdown().call()), 3)
+        assert.equal((await initialized.methods.enabledFlag().call()), true)
+
+        assert.equal((await initialized.methods.numVotesToShutdown().call()), 1)
```

Comment:
Another nit-picky comment, this looks like a copy and paste error. I also see this on line 261 and 369.